lib/mongo/server_selector/selectable.rb

```diff
@@ -154,8 +154,9 @@ def candidates(cluster)
         elsif cluster.sharded?
           near_servers(cluster.servers).each { |server| validate_max_staleness_support!(server) }
         else
-          validate_max_staleness_value!(cluster)
-          select(cluster.servers)
+          servers = cluster.servers
+          validate_max_staleness_value!(cluster) unless servers.empty?
```

Comment:
ReplicaSetNoPrimary/NoKnownServers.yml was added after my day ended yesterday so was not in this PR.
I've just updated them to the latest, was failing ReplicaSetNoPrimary/NoKnownServers.yml but then changed the code to do the check unless Topology type is Unknown.